bundles/org.vivoweb.dspacevivo.transformation/src/main/java/org/vivoweb/dspacevivo/transformation/harvester/oai/ItemItr.java

```diff
@@ -0,0 +1,75 @@
+package org.vivoweb.dspacevivo.transformation.harvester.oai;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import org.vivoweb.dspacevivo.model.Item;
+
+public class ItemItr implements Iterator<Item> {
```

Comment:
Not sure why do we need this class, but anyway its implementation should be improved. Beginning of hasNext is confusing, and the reason for that is overriding of next() method. List is also iterable, can we use next of that, and not removing elements from the list. Or it is implemented as that for the need of cleaning memory consumption (OAI supports resumption token). However, please consider redesigning of this class.